gen/plugin_test.go

```diff
@@ -169,6 +169,12 @@ func TestAddRootService(t *testing.T) {
 		}
 
 		g := newGenerateServiceBuilder(importer)
+
+		if spec.Parent != nil {
```

Comment:
@abhinav, I wonder if you feel this testing logic is sufficient. Here i'm manually replicating the logic by which addService adds the modules to the GenerateServiceBuilder. I'm particular, I'm ordering the addition of the module of parent first as the addService builder followed this logic before I moved the the AddModule logic out of AddService.

Reply:
This is fine.  Optionally, you can just mutate the Modules map directly.  The
contract from AddRootService's point of view is that every Thrift file has a
ModuleID entry.

At some point it would make sense to break apart the logic to fill
GenerateServiceRequests into more separate units rather than methods on the
same object.

Reply:
Right, I didn't mutate the modules map directly in order to exercise the functionality of the AddModules method. 👍 on the break apart logic step.